From 69b4170db97b21985ce1a72ab0fba2506e0f38f9 Mon Sep 17 00:00:00 2001
From: Frode Nordahl <fnordahl@ubuntu.com>
Date: Mon, 4 Nov 2024 12:04:27 +0100
Subject: [PATCH 16/31] northd: Sync routing data to pb.

From: Felix Huettner <felix.huettner@stackit.cloud>

this allows the the ovn-controller to later find all ports that
participate in dynamic routing.

Signed-off-by: Felix Huettner <felix.huettner@stackit.cloud>
---
 northd/northd.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/northd/northd.c b/northd/northd.c
index 75f1d652e..46e209fa4 100644
--- a/northd/northd.c
+++ b/northd/northd.c
@@ -4114,6 +4114,16 @@ sync_pb_for_lrp(struct ovn_port *op,
         }
     }
 
+    if (is_cr_port(op) || chassis_name) {
+        if (smap_get_bool(&op->nbrp->options, "maintain-vrf", false)) {
+            smap_add(&new, "maintain-vrf", "true");
+        }
+        if (smap_get_bool(&op->od->nbr->options, "dynamic-routing", false)) {
+            smap_add(&new, "dynamic-routing", "true");
+        }
+    }
+
+
     const char *ipv6_pd_list = smap_get(&op->sb->options, "ipv6_ra_pd_list");
     if (ipv6_pd_list) {
         smap_add(&new, "ipv6_ra_pd_list", ipv6_pd_list);
-- 
2.45.2

